simcon_project/conversation_templates/views/create_conversation_template.py

```diff
@@ -0,0 +1,50 @@
+import json
+from django.shortcuts import render
+from django.http import HttpResponse
+from django.contrib.auth.decorators import user_passes_test
+from django.views.decorators.csrf import ensure_csrf_cookie
+from conversation_templates.models import TemplateNodeChoice, TemplateNode, ConversationTemplate
+from users.models import Researcher
+
+def is_researcher(user):
+    return user.is_authenticated and user.is_researcher
+
+@user_passes_test(is_researcher)
+@ensure_csrf_cookie  # guarantees csrf cookie is generated even though html does not contain {% csrf_token %}
+def CreateConversationTemplateView(request):
+
+    if(request.method == 'GET'):
+        return render(request, 'template_management/create_conversation_template.html')
+
+    elif(request.method == 'POST'):
+        stored_nodes_dict = {}
+        body_unicode = request.body.decode('utf-8')
+        body = json.loads(body_unicode)
+
+        conv_temlpate = ConversationTemplate(
+            name=body['templateName'],
+            description=body['templateDescription'],
+            researcher=Researcher.objects.get(email=request.user.email)) #TODO Not sure if this is the best way to do it?
```

Comment:
Idk if there is a better way to do this... feels like there should be... like we should have direct access to the researcher we need since they are the one logged in (I just haven't done the hw on this so if someone knows, lmk).

Reply:
I think we can keep it as is for now. Once most our parts are hooked up I think we should go back and look into a way to just get the current researcher that is logged in, if there is a way to do it.

Reply:
I noticed that since "researcher" has to point to something on the Researcher table, a true superuser is not able to create any templates. Realizing now this might not cause an issue since we have the is_staff flag to denote our "app admin" but is something to be aware of I suppose. 
I think using the email to get the researcher is fine since it has to be unique. Using the id could also work -- id=request.user.id